evaluations/subcoin_2_piewol.md

```diff
@@ -0,0 +1,191 @@
+
+     test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
+
+          Running unittests src/lib.rs (target/debug/deps/subcoin_runtime_primitives-6e43d92ef6aec823)
+
+     running 0 tests
+
+     test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
+
+          Running unittests src/lib.rs (target/debug/deps/subcoin_service-0b18db2ae41f5b3f)
+
+     running 1 test
+     test block_executor::tests::off_runtime_in_memory_executor_should_produce_same_result_as_runtime_disk_executor ... ok
+
+     test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 36.71
+
+
+</details>
+
+## Docker
+Image is building. 
+
+<details>
+
+````
+ubuntu@ip-172-31-30-101:~/subcoin$ sudo docker build .
+[+] Building 1843.6s (12/12) FINISHED                            docker:default
+ => [internal] load build definition from Dockerfile                       0.1s
+ => => transferring dockerfile: 1.35kB                                     0.0s
+ => [internal] load metadata for docker.io/library/ubuntu:22.04            1.3s
+ => [internal] load .dockerignore                                          0.0s
+ => => transferring context: 2B                                            0.0s
+ => [internal] load build context                                        134.7s
+ => => transferring context: 12.66GB                                     134.6s
+ => [builder 1/5] FROM docker.io/library/ubuntu:22.04@sha256:340d9b015b19  3.5s
+ => => resolve docker.io/library/ubuntu:22.04@sha256:340d9b015b194dc6e2a1  0.0s
+ => => sha256:340d9b015b194dc6e2a13938944e0d016e57b967996 1.13kB / 1.13kB  0.0s
+ => => sha256:0eb0f877e1c869a300c442c41120e778db7161419244ee5 424B / 424B  0.0s
+ => => sha256:8a3cdc4d1ad3e314a91f76b7b99eed443f2152e3a9b 2.30kB / 2.30kB  0.0s
+ => => sha256:3713021b02770a720dea9b54c03d0ed83e03a2ef5 29.53MB / 29.53MB  1.2s
+ => => extracting sha256:3713021b02770a720dea9b54c03d0ed83e03a2ef5dce2898  2.0s
+ => [builder 2/5] WORKDIR /src                                           146.8s
+ => [builder 3/5] RUN apt-get update &&     DEBIAN_FRONTEND=noninteracti  43.5s
+ => [builder 4/5] COPY . .                                               176.3s 
+ => [builder 5/5] RUN /root/.cargo/bin/cargo build     --locked     --  1465.3s 
+ => [stage-1 2/3] COPY --from=builder /subcoin /subcoin                    0.2s 
+ => [stage-1 3/3] RUN mkdir /node-data && chown nobody:nogroup /node-data  0.2s 
+ => exporting to image                                                     0.3s 
+ => => exporting layers                                                    0.3s 
+ => => writing image sha256:9037f7d697a3625c15bc64fd85433e07bc5ebf0aebe98  0.0s
+ ````
+
+</details>
+
+## Docker
+
+Sadly I can't figure out how to start the node in docker. If I try to use `docker run <image_id> run` I get the following error. Any idea how I can make it work?
```

Comment:
I added a [commit](https://github.com/subcoin-project/subcoin/commit/b35da7ae2a4da76149de4988821f54455f692fb7) in the subcoin-milestone-2 branch that added the Docker command to run the node after submitting the delivery PR, can you try it again?

Reply:
I checked out that commit and build the image again. Sadly no luck. Please inform me about a way to successfully build and run the docker image so that the node is running.

```
$ git checkout b35da7a
Note: switching to 'b35da7a'.
HEAD is now at b35da7a Add Docker instruction for syncing the Bitcoin network

ubuntu@ip-172-31-30-101:~/subcoin$ sudo docker build .
[+] Building 7.2s (7/11)                                         docker:default
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 1.35kB                                     0.0s
 => [internal] load metadata for docker.io/library/ubuntu:22.04            1.3s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => [builder 1/5] FROM docker.io/library/ubuntu:22.04@sha256:adbb90115a21  2.3s
 => => resolve docker.io/library/ubuntu:22.04@sha256:adbb90115a21969d2fe6  0.0s
 => => sha256:075680e983398fda61b1ac59ad733ad81d18df4bc464116 424B / 424B  0.0s
 => => sha256:53a843653cbcd9e10be207e951d907dc2481d9c222d 2.30kB / 2.30kB  0.0s
 => => sha256:857cc8cb19c0f475256df4b7709003b77f101215e 29.54MB / 29.54MB  0.9s
 => => sha256:adbb90115a21969d2fe6fa7f9af4253e16d45f8d4c1 1.34kB / 1.34kB  0.0s
 => => extracting sha256:857cc8cb19c0f475256df4b7709003b77f101215ebf36931  1.2s
 => [internal] load build context                                          3.6s
 => => transferring context: 2.14MB                                        2.8s
 => [builder 2/5] WORKDIR /src                                             0.5s
 => CANCELED [builder 3/5] RUN apt-get update &&     DEBIAN_FRONTEND=noni  3.1s
ERROR: failed to solve: Canceled: context canceled
```

Reply:
Nothing special with the docker build. It's probably something (maybe network issue) that occurred on your side since you have succeeded in building the image locally before. I'd suggest you try again and see if it works.

In addition, you can still use the image built by GitHub Action by following the command in the commit to run the Docker, for https://github.com/subcoin-project/subcoin/commit/b35da7ae2a4da76149de4988821f54455f692fb7 contains no code changes to the subcoin-milestone-2 branch.

```bash
# For Docker users.
docker run -d -v data:/node-data --user root --name subcoin ghcr.io/subcoin-project/subcoin:v0.2.0 run -d /node-data --log subcoin_network=debug

# Check out the log.
docker logs -f subcoin
```